### PR TITLE
Adding Event feed

### LINKED ISSUE #102: Adding main page The Feed page
We need to add the main page of the app where we can see all the events.

### PR DESCRIPTION
**Pull request by Sendy459**
### What this PR does?

contains links for event creation and event info
feature to filter the events by category and location.
with styling .

### Notes for reviewer
ScreenshotEventFeed

### Related Issue

Resolves #102 

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes
- [ ] The changes in this PR do not require tests
- [x] The changes in this PR require tests that are not added yet

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

### CODE REVIEW

**Review by SaarOrus — APPROVED** [review]
Looks great!

File: static_home_page/templates/home_page/all_events.html
Comment (rotem2022) [inline static_home_page/templates/home_page/all_events.html:1]: pleas add a screenshot of the change

File: static_home_page/static/web_page_css.css [hunk 1]
```diff
@@ -127,3 +127,105 @@ p { font-size: 26px; }
...
+
+.form-button {
+  background-color: #e9bc52;
+  color: white;
+  cursor: pointer;
+}
+
+.form-button:hover {
+  background-color: #d1a843;
+}
```
Comment (Ofekmn) [inline static_home_page/static/web_page_css.css:231]: Please a newline at the end of file.

File: static_home_page/templates/home_page/all_events.html [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            <button class="EventButton"><a href="{% url 'view_event' user_id=user.id %}?id={{ event.id }}">Open Event</a></button>
+            </div>
+        </div>
+        
+    </div>
+    {% endfor %}
+
+{% endblock %}
+</body>
+</html>
```
Comment (Ofekmn) [inline static_home_page/templates/home_page/all_events.html:68]: Likewise. ^

Comment (rotem2022) [inline static_home_page/templates/home_page/all_events.html:68]: also here I think you should add a new line

**Review by Ofekmn — CHANGES_REQUESTED** [review]

File: static_home_page/templates/home_page/all_events.html [hunk 3]
```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (talko611) [inline static_home_page/templates/home_page/all_events.html:2]: This lines are not needed here you inherit this from the base.html

File: static_home_page/templates/home_page/all_events.html [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+{% extends 'base.html' %}
+{% block content %}
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Poppins&display=swap" rel="stylesheet">
+</head>
```
Comment (talko611) [inline static_home_page/templates/home_page/all_events.html:12]: You are entering an `<head>`  inside an `<body>` (This whole file is inside a body tag of the base.html file) are you sure this is the correct way to render the page?

File: static_home_page/views.py [hunk 5]
```diff
@@ -1,5 +1,32 @@
...
+    if request.method == "POST":
+        if 'Choose_filter' in request.POST:
+            filter_type = request.POST.get('Choose_filter')
+
+            if filter_type == 'Category' and 'Choose_Category' in request.POST:
+                category = request.POST.get('Choose_Category')
+                events = events.filter(category__name=category)
+            elif filter_type == 'Location' and 'Choose_Location' in request.POST:
+                location = request.POST.get('Choose_Location')
+                events = events.filter(location__name=location)
```
Comment (talko611) [inline static_home_page/views.py:22]: You can use the event manager search method to apply multiple filters in one call

**Review by rotem2022 — DISMISSED** [review]

File: fit_meet/urls.py [hunk 6]
```diff
@@ -16,12 +16,17 @@
...
+from event import views as event_views
 from django.contrib.auth import views as auth_views
 
 urlpatterns = [
     path('admin/', admin.site.urls),
     path('', views.static_home_page, name='static_home_page'),
     path('<int:user_id>/event/', include("event.urls")),
+    path('event_list/', views.event_list, name='event_list'),
+    path('create_event/<int:user_id>', event_views.create_event, name='create_event'),
+    path('<int:user_id>/event/info/', event_views.view_event, name='view_event'),
```
Comment (talko611) [inline fit_meet/urls.py:28]: You are creating here another path to the event views, which is wrong 
The event URLs that work with the event views functions 
should be in this pattern: `localhost:<port number>/<user id>/event/<name of the resource and query params if needed>` (<> are place holders and the text inside describes which value should be there)
This line `path('<int:user_id>/event/', include("event.urls"))` already tell django to look for matching URLs also event/urls.py
Instead, you should change the href attribute of the `<a>` to match the existing path defined in the event/url.py

**Review by talko611 — CHANGES_REQUESTED** [review]

File: fit_meet/urls.py (hunk 6 above)
Comment (Sendy459) [reply fit_meet/urls.py:28]: Thank you for your review ,
However, I'm having some trouble fully understanding the changes you're proposing. Could you please provide a specific example demonstrating how you'd like the urls path to be structured?
please clarify the changes needed in the href attribute and how it should match with the defined paths.

**Review by rotem2022 — APPROVED** [review]

**Review by talko611 — DISMISSED** [review]

**Review by Ofekmn — DISMISSED** [review]

File: event/views.py [hunk 7]
```diff
@@ -41,3 +43,27 @@ def view_event(request, user_id):
...
+                category = request.GET.get('Choose_Category')
+                events = events.filter(category__name=category)
+            elif filter_type == 'Location' and 'Choose_Location' in request.GET:
+                location = request.GET.get('Choose_Location')
+                events = events.filter(location__name=location)
+
+    if request.user.is_authenticated:
+        user_id = request.user.id
+    else:
+        user_id = None
```
Comment (kobihk) [inline event/views.py:64]: you can do it in one line like:
```suggestion
    user_id = request.user.id if request.user.is_authenticated else None
```

File: static_home_page/static/web_page_css.css [hunk 8]
```diff
@@ -131,6 +131,111 @@ p {
...
+
+.form-button:hover {
+  background-color: #d1a843;
```
Comment (kobihk) [inline static_home_page/static/web_page_css.css:238]: I think you missed the curly braces closure
```suggestion
  background-color: #d1a843;
  }
```

File: static_home_page/views.py [hunk 9]
```diff
@@ -1,5 +1,32 @@
...
+from event.models import Event
+from location.models import Location
+from category.models import Category
 
 
 def static_home_page(request):
     return render(request, 'home_page/index.html')
+
+
+def event_list(request):
```
[the remark this answers is not in the document]
Comment (kobihk) [reply static_home_page/views.py:11]: I can see that you already have it in:
event/views.py
Do we still need this code?

File: event/templates/event/all_events.html [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+                    Event</a></button>
+        </div>
+    </div>
+
+    </div>
+    {% endfor %}
+    {% endblock %}
+</body>
+
+</html>
```
Comment (kobihk) [inline event/templates/event/all_events.html:77]: you have it already in the base.html file
```suggestion

```

File: event/templates/event/all_events.html [hunk 11]
```diff
@@ -0,0 +1,77 @@
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Poppins&display=swap" rel="stylesheet">
+</head>
```
Comment (kobihk) [inline event/templates/event/all_events.html:8]: maybe all the "head" lines should be part of the base.html file

**Review by kobihk — CHANGES_REQUESTED** [review]

**Review by Ofekmn — APPROVED** [review]

**Review by rotem2022 — APPROVED** [review]

**Review by kobihk — APPROVED** [review]

**Pull request merged**